src/act.comm.cpp

```diff
@@ -100,17 +100,25 @@ ACMD(do_say)
         }
       }
     } else {
-      for (struct char_data *targ = get_ch_in_room(ch)->people; targ; targ = targ->next_in_room)
-        if (targ != ch && PLR_FLAGGED(targ, PLR_MATRIX) && !IS_IGNORING(targ, is_blocking_ic_interaction_from, ch)) {
-          // Send and store.
-          snprintf(buf, sizeof(buf), "Your hitcher says, \"%s^n\"\r\n", capitalize(arg_known_size));
-          send_to_char(buf, targ);
-          store_message_to_history(targ->desc, COMM_CHANNEL_SAYS, buf);
+      for (struct char_data *targ = get_ch_in_room(ch)->people; targ; targ = targ->next_in_room) {
+        if (targ == ch
+            || !PLR_FLAGGED(targ, PLR_MATRIX)
+            || !targ->persona
+            || !targ->persona->decker
+            || targ->persona->decker->hitcher != ch
+          ) {
+          continue;
         }
-      // Send and store.
-      snprintf(buf, sizeof(buf), "You send, down the line, \"%s^n\"\r\n", capitalize(arg_known_size));
-      send_to_char(buf, ch);
-      store_message_to_history(ch->desc, COMM_CHANNEL_SAYS, buf);
+        // We found our hitcher
+        snprintf(buf, sizeof(buf), "Your hitcher says, \"%s^n\"\r\n", capitalize(arg_known_size));
```

Comment:
Please add the `!IS_IGNORING(targ, is_blocking_ic_interaction_from, ch)` gag back to this. The hitcher should still think their message went through properly, but it should not print to the decker's screen.

Edit: I see that you shifted the ignore bit below to being handled when the hitcher connects, but there's also a case where someone starts ignoring a hitcher who is already connected.

Reply:
I didn't consider that and will do that.